### ISSUE TITLE
OAuth2 authorization problem - Argument 5 passed to AuthorizeController::__construct() must implement interface...

### ISSUE THREAD

**Issue by magician7** [mautic/mautic#11426]
### Mautic Version

4.4.x series

### PHP version

7.4.30

### What browsers are you seeing the problem on?

Chrome

### What happened?

When try to setup OAuth2 connection Mautic dosent respond properly. I got:
`{"errors":[{"message":"Looks like I encountered an error (error #500). If I do it again, please report me to the system administrator!","code":500,"type":null}]}`

### How can we reproduce this issue?

Step 1: 
Enable API connection in Mautic
Step 2:
Generate keys for OAuth2
Step 3:
Try to set up connection with those keys.

### Relevant log output

```shell
[2022-08-31 19:16:54] mautic.CRITICAL: Uncaught PHP Exception Symfony\Component\Debug\Exception\FatalThrowableError: "Argument 5 passed to FOS\OAuthServerBundle\Controller\AuthorizeController::__construct() must implement interface
Symfony\Component\Security\Core\Authentication\Token\Storage\TokenStorageInterface, instance of Symfony\Bundle\FrameworkBundle\Templating\DelegatingEngine given, called in /home/zycie/mautic_z/docroot/app/bundles/ApiBundle/Controlle
r/oAuth2/AuthorizeController.php on line 82" at /home/zycie/mautic_z/vendor/friendsofsymfony/oauth-server-bundle/Controller/AuthorizeController.php line 106 {"exception":"[object] (Symfony\\Component\\Debug\\Exception\\FatalThrowabl
eError(code: 0): Argument 5 passed to FOS\\OAuthServerBundle\\Controller\\AuthorizeController::__construct() must implement interface Symfony\\Component\\Security\\Core\\Authentication\\Token\\Storage\\TokenStorageInterface, instanc
e of Symfony\\Bundle\\FrameworkBundle\\Templating\\DelegatingEngine given, called in /home/zycie/mautic_z/docroot/app/bundles/ApiBundle/Controller/oAuth2/AuthorizeController.php on line 82 at /home/zycie/mautic_z/vendor/friendsofsym
fony/oauth-server-bundle/Controller/AuthorizeController.php:106)"} {"hostname":"server.xyz.com","pid":7061}
```

### Code of Conduct

- [X] I confirm that I have read and agree to follow this project's Code of Conduct

**Comment by mautibot** [comment]
This issue has been mentioned on **Mautic Community Forums**. There might be relevant details there:

https://forum.mautic.org/t/new-install-500-err-on-adding-plugins-using-basic-http-auth/25166/3

**Comment by trianity** [comment]
We have a too same error:

```
[2022-09-07 16:37:16] mautic.CRITICAL: Uncaught PHP Exception Symfony\Component\Debug\Exception\FatalThrowableError: "FOS\OAuthServerBundle\Controller\AuthorizeController::__construct(): Argument #5 ($tokenStorage) must be of type Symfony\Component\Security\Core\Authentication\Token\Storage\TokenStorageInterface, Symfony\Bundle\FrameworkBundle\Templating\DelegatingEngine given, called in /path-to-mautic/docroot/app/bundles/ApiBundle/Controller/oAuth2/AuthorizeController.php on line 82" at /path-to-mautic/vendor/friendsofsymfony/oauth-server-bundle/Controller/AuthorizeController.php line 106 {"exception":"[object] (Symfony\\Component\\Debug\\Exception\\FatalThrowableError(code: 0): FOS\\OAuthServerBundle\\Controller\\AuthorizeController::__construct(): Argument #5 ($tokenStorage) must be of type Symfony\\Component\\Security\\Core\\Authentication\\Token\\Storage\\TokenStorageInterface, Symfony\\Bundle\\FrameworkBundle\\Templating\\DelegatingEngine given, called in /path-to-mautic/docroot/app/bundles/ApiBundle/Controller/oAuth2/AuthorizeController.php on line 82 at /path-to-mautic/vendor/friendsofsymfony/oauth-server-bundle/Controller/AuthorizeController.php:106)"}
```
And unfortunately the linked forum by mautibot did not give any solution.

Mautic 4.4.1
PHP 8.0.19
Composer based installation

**Comment by MarkDaleman** [comment]
Hello,

I have resolved this issue by updating to the latest version (4.4.2). Prior I had this error on 4.4.1.

I updated with the following steps:

`sudo -u www-data php bin/console mautic:update:find`

`sudo -u www-data php bin/console mautic:update:apply`

`sudo -u www-data php bin/console mautic:update:apply --finish`

And to finish the update, clear the cache:

`sudo -u www-data php bin/console cache:clear --env=prod`

edit: also make sure that no cronjob for mautic is running atm!
